src/dfm/common/data_cleaning/dolma_taggers/language_scandi.py

```diff
@@ -0,0 +1,202 @@
+"""
+
+Filters.
+
+"""
+from collections.abc import Iterable
+
+import pycld2 as cld2
+import regex
+from anyascii import anyascii
+from dolma.core.data_types import DocResult, Document, Span, TextSlice
+from dolma.core.ft_tagger import BaseFastTextTagger, Prediction
+from dolma.core.registry import TaggerRegistry
+from dolma.core.taggers import BaseTagger
+from dolma.core.utils import split_paragraphs
+
+LANGS = {
+    "ENGLISH": "en",
+    "DANISH": "da",
+    "SWEDISH": "sv",
+    "NORWEGIAN": "no",
+    "ICELANDIC": "is",
+    "FAROESE": "fo",  # Note that FAROESE is not supported by cld2
+}
+
+
+@TaggerRegistry.add("cld2_scandi_doc")
+class Cld2LanguageFilterScandi(BaseTagger):
+    RE_BAD_CHARS = regex.compile(r"[\p{Cc}\p{Cs}]+")
```

Comment:
```suggestion
    RE_BAD_CHARS = regex.compile(r"[\p{Cc}\p{Cs}]+")
    # ﻿\p{Cc}: Matches a Unicode character with the "Other, Control" property, representing non-printing characters like newline, horizontal tab, escape, etc.
    # \p{Cs}: Matches a Unicode character with the "Other, Surrogate" property, which is used for representing characters from supplementary planes in UTF-16 (surrogate pairs).
```

Reply:
Do we want an `|` in between those two, or no?

Reply:
The square bracket expression `[\p{Cc}\p{Cs}]` is identical to `(\p{Cc}|\p{Cs})` and with the `+` it should match any sequence of 1 or more control characters.